### PR TITLE
Add methods to reset the Rasterizer without reallocating

### PR DESCRIPTION
**Pull request by rlespinet**

### CODE REVIEW

File: rasterizer/src/raster.rs [hunk 1]
```diff
@@ -45,6 +45,37 @@ impl Rasterizer {
...
+    /// Clears the rasterizer. This method behaves as if the Rasterizer were re-created with the same
+    /// dimensions, but does not perform an allocation.
+    ///
+    /// ```
+    /// use ab_glyph_rasterizer::Rasterizer;
+    /// let mut rasterizer = Rasterizer::new(14, 38);
+    /// rasterizer.clear();
+    /// ```
+    pub fn clear(&mut self) {
+        for px in self.a.iter_mut() {
```
Comment (alexheretic) [inline rasterizer/src/raster.rs:74]:
```suggestion
        for px in &mut self.a {
```
Minor: I think this is generally preferred when possible, looks like it produces less asm too...

File: rasterizer/src/raster.rs [hunk 2]
```diff
@@ -45,6 +45,37 @@ impl Rasterizer {
         }
     }
 
+    /// Resets the rasterizer to an empty `width` x `height` alpha grid. This method behaves as if
+    /// the Rasterizer were re-created, with the advantage of not allocating if the total number of
+    /// pixels of the grid does not increase.
+    ///
+    /// ```
+    /// use ab_glyph_rasterizer::Rasterizer;
```
Comment (alexheretic) [inline rasterizer/src/raster.rs:53]:
```suggestion
    /// # use ab_glyph_rasterizer::Rasterizer;
```

File: rasterizer/src/raster.rs [hunk 3]
```diff
@@ -45,6 +45,37 @@ impl Rasterizer {
...
+        self.height = height;
+        self.a.truncate(0);
+        self.a.resize(width * height + 4, 0.0);
+    }
+
+    /// Clears the rasterizer. This method behaves as if the Rasterizer were re-created with the same
+    /// dimensions, but does not perform an allocation.
+    ///
+    /// ```
+    /// use ab_glyph_rasterizer::Rasterizer;
```
Comment (alexheretic) [inline rasterizer/src/raster.rs:69]:
```suggestion
    /// # use ab_glyph_rasterizer::Rasterizer;
```

**Comment by alexheretic** [comment]
Looks good! Thanks for the pr & good documentation. I've added some very minor comments.

I assume this is to allow optimising allocations while rasterizing many glyphs on a single thread. Out of interest, which projects are you thinking of using these changes in?

File: rasterizer/src/raster.rs (hunk 1 above)
Comment (rlespinet) [reply rasterizer/src/raster.rs]: Thanks, I didn't know that! It seems that it indeed makes a difference at opt-level 0 and 1.

**Comment by rlespinet** [comment]
Thanks for the fast review, I'm using it to render fonts in a small project but my guess is that it could be used for other projects as well, since we usually know the maximum rendering size of a glyph for a given font and size.

**Pull request merged**